### ISSUE TITLE
In nested function `Tuple===Tuple` returns false 

### ISSUE THREAD

**Issue by mauro3** [JuliaLang/julia#11715]
On Julia 0.4.0-dev+5384:
```julia
check1(T)            = (@show "check1", T===Tuple; xdump(T))
check2{T}(::Type{T}) = (@show "check2", T===Tuple; xdump(T))
            
f{T}(::Type{T}) = (@show "f", T, T===Tuple ; check1(T); check2(T))
g(T)            = (@show "g", T, T===Tuple ; check1(T); check2(T))

check1(Tuple) # the === returns true
check2(Tuple) # the === returns true
g(Tuple) # here all === return true
f(Tuple) # here all === return true, except the one in check1
```
The two function calls return:
```
julia> f(Tuple) # here all === return true, except the one in check1
("f",T,T === Tuple) = ("f",Tuple,true)
("check1",T === Tuple) = ("check1",false)  # <---- ?
Tuple::DataType  <: Any
("check2",T === Tuple) = ("check2",true)
Tuple::DataType  <: Any

julia> g(Tuple) # here all === return true
("g",T,T === Tuple) = ("g",Tuple,true)
("check1",T === Tuple) = ("check1",true)
Tuple::DataType  <: Any
("check2",T === Tuple) = ("check2",true)
Tuple::DataType  <: Any
```
I thought I narrowed it down in #11712 but apparently that was another issue.

**Comment by JeffBezanson** [comment]
Ah, I see now.

**Comment by mauro3** [comment]
Yep, before the show-fix in #11712 some of the tuples were shown differently, thus my confusion.
